### PR TITLE
[Graphbolt] Convert a data block to dgl graphs

### PR DESCRIPTION
**Pull request by peizhou001**
## Description

Add a helper to convert data block to a dgl block.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
Please add unit test in this PR or a follow PR if we need to get it merged for example first.

**Comment by peizhou001** [comment]
Do not merge, need more tests.

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import dgl
+import dgl.graphbolt as gb
+import torch
+
+
+def test_to_dgl_graphs_hetero():
+    relation = ("A", "relation", "B")
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py:7]: worth to add features as well.

File: tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    g = gb.DataBlock(sampled_subgraphs=[subgraph]).to_dgl_graphs()[0]
+    assert torch.equal(g.edges()[0], node_pairs[relation][0])
+    assert torch.equal(g.edges()[1], node_pairs[relation][1])
+    assert torch.equal(g.ndata[dgl.NID]["A"], reverse_row_node_ids["A"])
+    assert torch.equal(g.ndata[dgl.NID]["B"], reverse_row_node_ids["B"])
+    assert torch.equal(g.edata[dgl.EID], reverse_edge_ids[relation])
+
+
+def test_to_dgl_graphs_homo():
+    node_pairs = (torch.tensor([0, 1, 2]), torch.tensor([0, 4, 5]))
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py:31]: ditto

File: python/dgl/graphbolt/data_block.py [hunk 3]
```diff
@@ -84,6 +84,7 @@ def to_dgl_graphs(self):
                         edge_type,
                         feature_name,
                     ), feature in edge_feature.items():
+                        edge_type = tuple(edge_type.split(":"))
```
Comment (frozenbugs) [inline python/dgl/graphbolt/data_block.py:87]: use etype_tuple_to_str, added in https://github.com/dmlc/dgl/pull/6230

Comment (frozenbugs) [reply python/dgl/graphbolt/data_block.py:87]: So tuple is used in DGLGraph?

Comment (Rhett-Ying) [reply python/dgl/graphbolt/data_block.py:87]: yes

File: tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import dgl
+import dgl.graphbolt as gb
+import torch
+
+
+def test_to_dgl_graphs_hetero():
+    relation = ("A", "relation", "B")
+    node_pairs = {relation: (torch.tensor([0, 1, 2]), torch.tensor([0, 4, 5]))}
```
[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py:8]: If this PR could be pending, I suggest to update this PR after all etypes are converted to str in GB.

[the remark this answers is not in the document]
Comment (peizhou001) [reply tests/python/pytorch/graphbolt/impl/test_to_dgl_graphs.py:8]: This PR can be postponed after refactoring.

**Pull request merged**